### ISSUE TITLE
Changing maximum frequency makes  Windows unusable

### ISSUE THREAD

**Issue by johkra** [mpollice/AmdMsrTweaker#1]
First of all: Thanks a lot for updating this tool for Trinity! I had issues with AMD Overdrive and my Wifi/Bluetooth and this tool is a godsend.

To the problem: Changing P2 (the maximum non-boosted frequency) leads to Windows losing the graphical output and only showing a black screen in low resolution. It is still running as by pressing Ctrl+Alt+Entf I can shortly see a lock screen (blue, with my user name and the icon above) also in very low resolution.

Changing any of P3-P7 does not have this effect. Changing the multiplicator in BIOS works and changes P2.

I believe this is an issue with Windows, but I can't confirm this because I have only one machine on which to test.

Should it be a Windows issue, then I would suggest to add a Warning message. I can submit a patch for this.

My configuration:
A10-5700
MSI FM2-A75IA-E53
Windows 8 Pro 64 bit

**Comment by mpollice** [comment]
I was able to kind of reproduce the problem on Windows 7 too. The thing is on Windows 7 the screen just flickers and sometimes you lose the Aero style, but other than that you can continue to work and even launch 3D applications. Due to the new UI on Win8, there the problem is more pronounced.

I tested this on a 5800K on a ASUS F2A85-M. I know of another user who has the same issue on a 5700 and some ASRock board. Furthermore the same problem seems to be present on AM3+ with various other CPUs. Similar problems arise also with AMDs own Overdrive tool.

At this point I don't really understand why it happens other than it does happen. Sure warning the user would be a fix as long as we don't know what is the cause and how to avoid it.

**Comment by johkra** [comment]
I'll try to send a patch/pull request tomorrow.

I also found a workaround: In the advanced options of the energy plan (balanced) I changed the maximum CPU speed to 90%. This prevents the CPU from entering P2 and I can set up P3-P7 as I want. Contrary to fixing the multiplicator in the BIOS this does not disable Cool'n'Quiet. I have turbo disabled and have not tested how this changes turbo behaviour.

**Comment by mpollice** [comment]
Well, but setting P3-P7 works fine either way so what do you accomplish?

I am currently discussing this issue with users in the hardwareluxx.de forum. You may send a patch but you don't need to. Since a lot of users are affected by the problem I will add a warning that informs the user about possible problems and asks for confirmation of setting the multiplier for P2, as you suggested.

**Comment by johkra** [comment]
My goal is to change the maximum frequency while keeping Cool'n'Quiet. Specifically, I want to downclock (in order to heavily undervolt) my APU to 2.8GHz. If I do this in the BIOS, the CPU will always run at the set frequency. I can manually change P-States, but Cool'n'Quiet does not work.

What I accomplish is thus that the P2 state is completely ignored and I can change the multiplicator however I want without running into the issues from the first post. In other words, my effective range is 2800-800MHz, Cool'n'Quiet works and I get no crash of the graphical interface.

**Comment by powerrush** [comment]
bcdedit /set useplatformclock true
bcdedit /set tscsyncpolicy Enhanced
bcdedit /set disabledynamictick yes

This is the solution....

**Comment by powerrush** [comment]
I am writing a program in c# , a user interface for this program, and a core temperature control. I want to control my cpu in order to degrade performance if the limit temperature is achieved.

**Comment by powerrush** [comment]
i have my fx 8320 overclocked to 4.8 ghz, and depending on temperature it downclocks to another pstate

**Comment by mpollice** [comment]
Aside from this discussion being a bit misplaced here, I wonder what you try to accomplish. You want to make it downclock after it exceeds a certain temperature threshold?

**Comment by powerrush** [comment]
Yes i have done that, i'm using your software for setting the pstates at the time the service loads, and it downclocks after it exceeds a certain temperature. It's working and has a few surprises because, single threading is better now than before, games runs with less fps drops, and less power consumption. I am using the powrprof.dll library of windows in order to change the cpu performance state. Maybe there is a way of using winring0 to force the processor to enter in another pstate? I don't know, winring0 is so difficult for me...

**Comment by powerrush** [comment]
pd: it's a kind of cpu governor, and not a Energy plan.

**Comment by powerrush** [comment]
Hi, is in the news now:
http://wccftech.com/amd-ryzen-desktop-zen-cpu-xfr/

It’s all thanks to a new technology that the company calls “XFR”, short for Extended Frequency Range. Which automatically boosts the clock speed of RYZEN CPUs beyond their official nominal values and stretch them as far as the cooling will allow, without any intervention from the user.

It is hardware + software (bios-microcode). My program does the same thing. It overclocks on the need, and downclocks if the max temperature is reached.
